plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/service/QueryGroupPersistenceService.java

```diff
@@ -192,6 +194,21 @@ private Map<ResourceType, Double> calculateTotalUsage(Map<String, QueryGroup> ex
         return map;
     }
 
+    /**
+     * Get the QueryGroups with the specified name from cluster state
+     * @param name - the QueryGroup name we are getting
+     * @param currentState - current cluster state
+     */
+    public static List<QueryGroup> getFromClusterStateMetadata(String name, ClusterState currentState) {
+        Map<String, QueryGroup> currentGroups = currentState.getMetadata().queryGroups();
+        if (name == null || name.isEmpty()) {
+            return new ArrayList<>(currentGroups.values());
+        }
+        List<QueryGroup> resultGroups = new ArrayList<>();
+        currentGroups.values().stream().filter(group -> group.getName().equals(name)).findFirst().ifPresent(resultGroups::add);
+        return resultGroups;
```

Comment:
return `currentGroups.values().stream().filter(group -> group.getName().equals(name)).findAny().stream().toArray();`. Having `findAny()` instead of `findFirst()` allows parallelization, although might not matter as much since our list should be fairly small due to limitations on the number of QueryGroups.